contracts/tequila-train/tests/test.rs

```diff
@@ -1,9 +1,110 @@
-use gstd::ActorId;
+use gstd::{ActorId, Encode};
 use gtest::{Program, System};
 use tequila_train_io::*;
 
+pub trait TestFunc {
+    fn register(&self, from: u64, player: ActorId, name: String, error: Option<Error>);
+    fn start_game(&self, from: u64, error: Option<Error>);
+    fn restart_game(&self, from: u64, players_limit: Option<u64>, error: Option<Error>);
+    fn skip(&self, from: u64, error: Option<Error>);
+    fn place(
+        &self,
+        from: u64,
+        tile_id: u32,
+        track_id: u32,
+        remove_train: bool,
+        error: Option<Error>,
+    );
+}
+
+impl TestFunc for Program<'_> {
+    fn register(&self, from: u64, player: ActorId, name: String, error: Option<Error>) {
+        let result = self.send(
+            from,
+            Command::Register {
+                player,
+                name: name.clone(),
+            },
+        );
+        assert!(!result.main_failed());
+        let reply: Result<Event, Error>;
+        if let Some(error) = error {
+            reply = Err(error);
+            assert!(result.contains(&(from, reply.encode())));
+        } else {
+            reply = Ok(Event::Registered { player, name });
+            assert!(result.contains(&(from, reply.encode())));
+        }
```

Comment:
It might be rewritten in func style, but at least use if-expression.

```suggestion
        let reply = if let Some(error) = error {
            Err(error)
        } else {
            Ok(Event::Registered { player, name })
        }
        assert!(result.contains(&(from, reply.encode())));
```